Dockerfile

```diff
@@ -0,0 +1,27 @@
+##
+## build with:
+## docker buildx build --platform=local --output dist .
+##
+## binary for your arch will be inside dist folder
+##
+## to build for OSX, run previously:
+## docker buildx create --use --platform darwin/amd64
+##
+
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:golang AS xgo
```

Comment:
Why do you need this?

Reply:
https://github.com/tonistiigi/xx/tree/master/golang
it's used to do cross build... I need that to build the mac bin on mac

Reply:
otherwise it builds a linux bin

Reply:
Resolved by mistake.

Go can do cross compilation natively with GOOS and GOARCH. I would drop this dep and just rely on go tooling.

Reply:
when building on a mac it uses a linux vm, which then the arch is wrong for the built artifacts
this pushes go to use the right arch

Reply:
Ok got it.